### PR TITLE
dev-python/apache-airflow/:Pushing my attempt at an ebuild for Apache…

### PR DESCRIPTION
**Pull request by MCMXCIII**
… Airflow.

This is my attemt at an Ebuild I have based it off the requirements.txt for Airflow

which is below.

```
click>=8.0
cached_property;python_version<"3.8"
jinja2>=2.10
keyring==10.1
PyGithub
jsonpath_ng
jsonschema
pendulum
pyyaml
packaging
rich
rich_click
semver
tabulate
```

which can be found here:https://github.com/apache/airflow/blob/main/dev/requirements.txt

### CODE REVIEW

File: dev-python/apache-airflow/apache-airflow.0.0.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	>=dev-python/jinja-3.1.2
+	>=dev-python/keyring-23.6.0
+	>=dev-python/PyGithub-1.55-r1
+	>=dev-python/jsonschema-4.7.2
+	>=dev-python/pendulum-2.1.2-r1
+	>=dev-python/pyyaml-6.0-r1
+	>=dev-python/packaging-21.3-r3
+	>=dev-python/rich-12.5.1
+	>=dev-python/semver-2.13.0
+	>=dev-python/0.8.10"
```
Comment (alip) [inline dev-python/apache-airflow/apache-airflow.0.0.1.ebuild:33]: package name missing here.

Comment (MCMXCIII) [reply dev-python/apache-airflow/apache-airflow.0.0.1.ebuild]: Should be fixed now. Sorry about that.

**Review by alip — COMMENTED** [review]
The ebuild should be named `apache-airflow-0.0.1.ebuild` or else ebuild complains:

`ebuild: /home/alip/src/gentoo-overlay/dev-python/apache-airflow/apache-airflow.0.0.1.ebuild: dev-python/apache-airflow.0.0.1: does not follow correct package syntax`

**Comment by alip** [comment]
I pushed a few changes to make it build, but we still have some runtime dependencies missing, such as [dev-python/colorlog](https://packages.gentoo.org/packages/dev-python/colorlog) which is already packaged and [lazy_object_proxy](https://pypi.org/project/lazy-object-proxy/) which is not packaged.

**Comment by alip** [comment]
Quoting [Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/start/local.html) about installation:
```
# Install Airflow using the constraints file
AIRFLOW_VERSION=2.3.4
PYTHON_VERSION="$(python --version | cut -d " " -f 2 | cut -d "." -f 1-2)"
# For example: 3.7
CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}> /constraints-${PYTHON_VERSION}.txt"
# For example: https://raw.githubusercontent.com/apache/airflow/constraints-2.3.4/constraints-3.7.txt
pip install "apache-airflow==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"
```

The constraints url has a list of packages.
Meanwhile we could go for 2.3.4 instead of 2.3.3 which has been released.

**Comment by alip** [comment]
> ... and [lazy_object_proxy](https://pypi.org/project/lazy-object-proxy/) which is not packaged.

Correction, this is also packaged with the name [lazy-object-proxy](https://packages.gentoo.org/packages/dev-python/lazy-object-proxy).

**Comment by alip** [comment]
What's left to be done:
1. A few more dependencies need to be packaged (from install_requires in setup.cfg): 
  * [connexion[swagger-ui,flask]>=2.10.0](https://pypi.org/project/connexion2/)
  * [flask-appbuilder==4.1.2](https://pypi.org/project/Flask-AppBuilder/)
  * [flask-caching>=1.5.0](https://pypi.org/project/Flask-Caching/)
  * [flask-session>=0.4.0](https://pypi.org/project/Flask-Session/)
  * [linkify-it-py>=2.0.0](https://pypi.org/project/linkify-it-py/)
  * [marshmallow-oneofschema>=2.0.1](https://pypi.org/project/marshmallow-oneofschema/)
  * [sqlalchemy_jsonfield>=1.0](https://pypi.org/project/SQLAlchemy-JSONField/)
2. The ebuild should create airflow system user and group via respective packages in acct-{user,group}
3. Add an initial init script and a conf.d configuration file.

**Comment by alip** [comment]
All the missing core dependencies have been packaged. With that `airflow standalone` works. Airflow has many providers that come as additional packages. We want to package the postgres provider to begin with.

**Comment by alip** [comment]
Postgres provider is packaged and the ebuild now creates a system user and group for airflow.
`airflow standalone` is meant for development use only, for production we want to create init scripts for various airflow services. As reference [the systemd services shipped with airflow](https://github.com/apache/airflow/tree/main/scripts/systemd) may be used.

**Comment by alip** [comment]
Added more dependencies and wrote init scripts for airflow services.
This should be good enough for the first iteration, we can add more optional dependencies in the future as necessary.

**Comment by MCMXCIII** [comment]
Waiting on a review.

**Review by monsieurp — APPROVED** [review]
👍

**Pull request merged**